tests/helpers.bash

```diff
@@ -141,36 +141,34 @@ function wait_for_k8s_endpoints {
 }
 
 function wait_for_cilium_status {
-    set +x
-	while ! cilium status; do
-	    micro_sleep
-	done
+  local NUM_DESIRED="1"
+  local CMD="cilium status | grep 'Cilium:' | grep -c OK || true"
+  local INFO_CMD="true"
+  local MAX_MINS="1"
+  local ERROR_OUTPUT="Timeout while waiting for Cilium to be ready"
+  wait_for_desired_state "$NUM_DESIRED" "$CMD" "$INFO_CMD" "$MAX_MINS" "$ERROR_OUTPUT"
 }
 
 function wait_for_kubectl_cilium_status {
-    set +x
-    namespace=$1
-    pod=$2
-
-    echo "Waiting for Cilium to spin up"
-    while ! kubectl -n ${namespace} exec ${pod} cilium status; do
-        micro_sleep
-    done
-    set -x
+  set +x
+  check_num_params "$#" "2"
+  namespace=$1
+  pod=$2
+  local NUM_DESIRED="1"
+  local CMD="kubectl -n ${namespace} exec ${pod} cilium status | grep "Cilium:" | grep -c 'OK' || true"
+  local INFO_CMD="true"
+  local MAX_MINS="2"
+  local ERROR_OUTPUT="Timeout while waiting for Cilium to be ready"
+  wait_for_desired_state "$NUM_DESIRED" "$CMD" "$INFO_CMD" "$MAX_MINS" "$ERROR_OUTPUT"
 }
 
 function wait_for_cilium_ep_gen {
-    set +x
-    while true; do
-        # FIXME by the time this executed, it's not guaranteed that we
-        # don't skip a regenerating
-        sleep 2s
-        if ! cilium endpoint list | grep regenerating; then
-            break
-        fi
-        micro_sleep
-    done
-    set -x
+  local NUM_DESIRED="0"
+  local CMD="cilium endpoint list | grep -c regenerating"
```

Comment:
We need to keep a FIXME comment in here as this remains unreliable. Ideally with a GH-XXX reference to a created issue